content/blog/2019-11-02-custom-alloc/index.md

```diff
@@ -0,0 +1,206 @@
+
+On the other hand, custom memory allocation is not quite dead. [Here](https://github.com/mtrebi/memory-allocators)'s a tutorial on how and why custom allocators are helpful, custom allocation still is seen as a potential reason to disregard projects such as [Mesh][], and Emry Berger's [Heap Layers][heapl] project (used to construct reaps in the present paper), which has been much more widely employed, ironically is a framework for building custom allocators, by its own admission:
+
+> "Heap Layers makes it easy to write high-quality custom and general-purpose memory allocators."
+
+
+**What happened to reaps?**
+Though modern memory allocators more effectively put off frees to make them more region-like, "reap" is not a common term for memory allocators --- in fact, only a small handful of the papers that cite this one mention reaps (and all share authors). 
+
+It is interesting that the reap, which one might consider the more substantive, creative contribution to this paper, does not seem to have caught on. In his [2012 blog post](https://emeryblogger.com/2012/10/28/most-influential-oopsla2012/) in which he acknowledges a most influential paper award for this paper, Emery tells people (with extreme brevity and no explanation) to use [his more modern allocator, Hoard](hoard.org), instead of reaps. Even more strangely [the technical paper] that hoard is based on does not include the word "reap", leaving us wondering what happened to them. Still, an implementation lives on in [the legacy portion](https://github.com/emeryberger/Heap-Layers/blob/master/legacy/reap/regionheap.cpp) of the Heap Layers project.
+
+# Conclusions
+
+While this paper seems to have been largely correct in several important ways, many of its elements leave something to be desired, as discussed in the [Empirical Setup section](#experimental-setup). The evaulation seems thorough but unprinicpled and worrylingly oversteps the boundaries of good empiricism with its conclusions; the new algorithm and data structure (reaps) seem very useful and are sold well, but explained phenominally poorly, with only a partial example which doesn't illustrate interesting cases, no theoretical backing, and no explanation of the heuristics that enable its performance. 
+
+These reaps do not seem to have been used more than once or twice since 2002, and not for lack of publicity (several claim, nebulously, to use "ideas from reaps"). It is unclear whether the reason they were not adopted was because of the poor explantion, to some fundamental flaw which was later discovered, or because they never were quite as good as they look (after all, there are only 8 data points, and it might be easy to overfit to them with parameter or design tuning). With no follow-up studies or testing on benchmarks for which they were not designed, it's hard to know.
```

Comment:
Once more, I think you're a little too harsh—and needlessly so—when you invoke "poor explanation." Just say it's not clear why they haven't been adopted and you'll avoid the appearance of writing a put-down.

Reply:
I agree, I've re-worded it and changed the order, but kept the reasons. The intention is for this paragraph to be mostly negative (but not an attack), and softened by the final one.